### PR TITLE
Allowing downloadable checksums and different checksum formats

### PR DESCRIPTION
**Pull request by huutomerkki**
This PR changes most hardcoded checksums in this repo to downloadable ones. The exception is krew as there is no checksum available.  Support for checksums provided in a checksums.txt file is also added. 

Also renaming some variables so that if need comes, it'll be easier to add different checksum formats.

### CODE REVIEW

**Review by Rozzii — COMMENTED** [review]
/lgtm

File: lib/download.sh [hunk 1]
```diff
@@ -51,20 +55,31 @@ wget_and_verify()
...
     wget "${args[@]}"
 
-    if [[ "${sha256}" =~ https ]]; then
-        sha256="$(curl -SsL "${sha256}")"
+    if [[ "${checksum}" =~ https ]]; then
+        checksum="$(curl -SsL "${checksum}")"
+    fi
+
+    if [[ "${checksum}" =~ ${dlname,,} ]]; then
+        while read -r line
```
Comment (tuminoid) [inline lib/download.sh:63]:
```suggestion
        while read -r line; do
```

File: lib/download.sh [hunk 2]
```diff
@@ -51,20 +55,31 @@ wget_and_verify()
...
+        do
```
Comment (tuminoid) [inline lib/download.sh:64]:
```suggestion
```

**Review by tuminoid — COMMENTED** [review]
A minor style nit, otherwise looks solid improvement.

Also,
/hold
for a sec to let me test the various scenarios. This seems to handle both direct and multi-arch files, so expecting a quick pass.

File: lib/download.sh [hunk 3]
```diff
@@ -142,6 +159,7 @@ download_and_install_kind()
 download_and_install_tilt()
 {
     TILT_URL="https://raw.githubusercontent.com/tilt-dev/tilt/${TILT_VERSION}/scripts/install.sh"
+    TILT_SHA256="${TILT_SHA256:-https://github.com/tilt-dev/tilt/releases/download/${TILT_VERSION}/checksums.txt}"
```
Comment (tuminoid) [inline lib/download.sh:162]:
```suggestion
    TILT_SHA256="${TILT_SHA256:-https://github.com/tilt-dev/tilt/releases/download/${TILT_VERSION}/checksums.txt}"
```

File: lib/download.sh [hunk 4]
```diff
@@ -128,6 +144,7 @@ download_and_install_kvm2_driver()
 download_and_install_kind()
 {
     KIND_URL="https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-$(uname)-amd64"
+    KIND_SHA256="${KIND_SHA256:-https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-$(uname)-amd64.sha256sum}"
```
Comment (tuminoid) [inline lib/download.sh:147]:
```suggestion
    KIND_SHA256="${KIND_SHA256:-https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-$(uname)-amd64.sha256sum}"
```

File: lib/download.sh [hunk 5]
```diff
@@ -51,20 +55,30 @@ wget_and_verify()
     fi
     wget "${args[@]}"
 
-    if [[ "${sha256}" =~ https ]]; then
-        sha256="$(curl -SsL "${sha256}")"
+    if [[ "${checksum}" =~ https ]]; then
+        checksum="$(curl -SsL "${checksum}")"
+    fi
+
+    if [[ "${checksum}" =~ ${dlname,,} ]]; then
```
Comment (tuminoid) [inline lib/download.sh:62]: This does not work for Tilt. 

1. We download Tilt's installation script, and the checksum is for release tarball. It cannot find install.sh on the checksum file, and skips it.
2. It does not return failure for this case, and happily goes and installs TIlt anyways.

We either need to change the download itself to the release tarball or use hardcoded SHA for the script. Using hardcoded SHA wouldn't be too bad, as it is the same script regardless of arch.

**Review by tuminoid — COMMENTED** [review]
Whitespace nits + actual issues with Tilt verification.

File: lib/download.sh (hunk 5 above)
Comment (huutomerkki) [reply lib/download.sh:62]: Ah, I had missed that we use a different install method for tilt. I can change it back to the pinned SHA for now, but changing the installation method shouldn't be too hard either if we want that instead.

File: lib/download.sh [hunk 6]
```diff
@@ -142,7 +159,7 @@ download_and_install_kind()
 download_and_install_tilt()
 {
     TILT_URL="https://raw.githubusercontent.com/tilt-dev/tilt/${TILT_VERSION}/scripts/install.sh"
-    TILT_SCRIPT="install.sh"
+   TILT_SCRIPT="install.sh"
```
Comment (tuminoid) [inline lib/download.sh:162]:
```suggestion
    TILT_SCRIPT="install.sh"
```

**Review by tuminoid — COMMENTED** [review]
One whitespace mishap, but this is good to merge otherwise.

/approve
/hold for the whitespace fix

**Pull request merged**